app/views/core/CreateAccountModal/BasicInfoView.coffee

```diff
@@ -0,0 +1,265 @@
+CocoView = require 'views/core/CocoView'
+AuthModal = require 'views/core/AuthModal'
+template = require 'templates/core/create-account-modal/basic-info-view'
+forms = require 'core/forms'
+errors = require 'core/errors'
+User = require 'models/User'
+State = require 'models/State'
+
+###
+This view handles the primary form for user details — name, email, password, etc,
+and the AJAX that actually creates the user.
+
+It also handles facebook/g+ login, which if used, open one of two other screens:
+sso-already-exists: If the facebook/g+ connection is already associated with a user, they're given a log in button
+sso-confirm: If this is a new facebook/g+ connection, ask for a username, then allow creation of a user
+
+The sso-confirm view *inherits from this view* in order to share its account-creation logic and events.
+This means the selectors used in these events must work in both templates.
+
+This view currently uses the old form API instead of stateful render.
+It needs some work to make error UX and rendering better, but is functional.
+###
+
+module.exports = class BasicInfoView extends CocoView
+  id: 'basic-info-view'
+  template: template
+
+  events:
+    'change input[name="email"]': 'onChangeEmail'
+    'change input[name="name"]': 'onChangeName'
+    'click .back-button': 'onClickBackButton'
+    'submit form': 'onSubmitForm'
+    'click .use-suggested-name-link': 'onClickUseSuggestedNameLink'
+    'click #facebook-signup-btn': 'onClickSsoSignupButton'
+    'click #gplus-signup-btn': 'onClickSsoSignupButton'
+
+  initialize: ({ @signupState } = {}) ->
+    @state = new State {
+      suggestedNameText: '...'
+      checkEmailState: 'standby' # 'checking', 'exists', 'available'
+      checkEmailValue: null
+      checkEmailPromise: null
+      checkNameState: 'standby' # same
+      checkNameValue: null
+      checkNamePromise: null
+      error: ''
+    }
+    @listenTo @state, 'change:checkEmailState', -> @renderSelectors('.email-check')
+    @listenTo @state, 'change:checkNameState', -> @renderSelectors('.name-check')
+    @listenTo @state, 'change:error', -> @renderSelectors('.error-area')
+    @listenTo @signupState, 'change:facebookEnabled', -> @renderSelectors('.auth-network-logins')
+    @listenTo @signupState, 'change:gplusEnabled', -> @renderSelectors('.auth-network-logins')
+    
+  afterRender: ->
+    super()
+    if suggestedName = @state.get('suggestedName')
+      @setNameError(suggestedName)
+
+  onChangeEmail: ->
+    @checkEmail()
+    
+  checkEmail: ->
+    email = @$('[name="email"]').val()
+    if email is @state.get('lastEmailValue')
+      return @state.get('checkEmailPromise')
+      
+    if not email
+      @state.set({
+        checkEmailState: 'standby'
+        checkEmailValue: email
+        checkEmailPromise: null
+      })
+      return
```

Comment:
Not returning a promise some of the time feels a bit fragile — I have to look up places that call this function and check their early-exit logic to see if they would ever try to erroneously call `.then` on it. I assume this is fine because onSubmitForm calls checkBasicInfo first, but it's good to keep that kind of uncertainty low.